src/Uno.UWP/Globalization/NumberFormatting/CurrencyFormatter.cs

```diff
@@ -0,0 +1,175 @@
+	{
+		_formatterHelper = new FormatterHelper();
+		_translator = new NumeralSystemTranslator();
+
+		var currencyData = CurrencyData.GetCurrencyData(currencyCode);
+
+		if (currencyData == CurrencyData.Empty)
+		{
+			ExceptionHelper.ThrowArgumentException(nameof(currencyCode));
+		}
+
+		_currencyData = currencyData;
+		FractionDigits = currencyData.DefaultFractionDigits;
+		_startWithInCurrencyCodeMode = $"{_currencyData.CurrencyCode}{NoBreakSpaceChar}";
+	}
+
+	public CurrencyFormatterMode Mode { get; set; }
+
+	public bool IsDecimalPointAlwaysDisplayed { get => _formatterHelper.IsDecimalPointAlwaysDisplayed; set => _formatterHelper.IsDecimalPointAlwaysDisplayed = value; }
+
+	public int IntegerDigits { get => _formatterHelper.IntegerDigits; set => _formatterHelper.IntegerDigits = value; }
+
+	public bool IsGrouped { get => _formatterHelper.IsGrouped; set => _formatterHelper.IsGrouped = value; }
+
+	public string NumeralSystem
+	{
+		get => _translator.NumeralSystem;
+		set => _translator.NumeralSystem = value;
+	}
+
+	public IReadOnlyList<string> Languages => _translator.Languages;
+
+	public string ResolvedLanguage => _translator.ResolvedLanguage;
+
+	public int FractionDigits { get => _formatterHelper.FractionDigits; set => _formatterHelper.FractionDigits = value; }
+
+	public INumberRounder? NumberRounder { get; set; }
+
+	public bool IsZeroSigned { get => _formatterHelper.IsZeroSigned; set => _formatterHelper.IsZeroSigned = value; }
+
+	public int SignificantDigits { get => _formatterHelper.SignificantDigits; set => _formatterHelper.SignificantDigits = value; }
+
+	public string Format(double value) => FormatDouble(value);
+
+	public string FormatDouble(double value)
+	{
+		if (!_formatterHelper.TryValidate(value, out string text))
+		{
+			return text;
+		}
+
+		if (NumberRounder != null)
+		{
+			value = NumberRounder.RoundDouble(value);
+		}
+
+		bool needParentheses = false;
+		var stringBuilder = StringBuilderPool.Instance.Get();
+
+		switch (Mode)
+		{
+			case CurrencyFormatterMode.UseSymbol:
+				stringBuilder.Append(_currencyData.Symbol);
+				break;
+			case CurrencyFormatterMode.UseCurrencyCode:
+				stringBuilder.Append(_currencyData.CurrencyCode);
+				stringBuilder.Append(NoBreakSpaceChar);
+				break;
+		}
+
+		if (value == 0d)
+		{
+			if (IsZeroSigned && value.IsNegative())
+			{
+				_formatterHelper.AppendFormatZero(stringBuilder);
+				needParentheses = true;
+			}
+			else
+			{
+				_formatterHelper.AppendFormatZero(value, stringBuilder);
+			}
+		}
+		else
+		{
+			if (value.IsNegative())
+			{
+				value = -value;
+				needParentheses = true;
+			}
+
+			_formatterHelper.AppendFormatDouble(value, stringBuilder);
+		}
+
+		_translator.TranslateNumerals(stringBuilder);
+
+		if (needParentheses)
+		{
+			stringBuilder.Insert(0, '(');
+			stringBuilder.Append(')');
+		}
+
+		var formatted = stringBuilder.ToString();
+
+		StringBuilderPool.Instance.Return(stringBuilder);
+		return formatted;
+	}
```

Comment:
@workgroupengineering I created simple StringBuilderPool. Also I converted more method to use StringBuilder. I'll be happy to hear from you.

Reply:
Sorry for the delay, I've been busy.